scripts/docs-csat.py

```diff
@@ -0,0 +1,318 @@
+                        {'startDate': args.start_date, 'endDate': args.end_date}
+                    ],
+                    'metrics': [
+                        {'expression': 'ga:uniquePageviews'},
+                    ],
+                    'dimensions': [
+                        {'name': 'ga:pageTitle'},
+                        {'name': 'ga:pagePath'}
+                    ],
+                    'dimensionFilterClauses': [
+                        {
+                            'filters': [
+                                {
+                                    'dimensionName': 'ga:pagePath',
+                                    'operator': 'IN_LIST',
+                                    'expressions': [pages]
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+    ).execute()
+
+
+no_votes = dict()
+yes_votes = dict()
+pageviews = dict()
+def create_dict(response, dict):
+    for report in response.get('reports', []):
+
+        for row in report.get('data', {}).get('rows', []):
+            page = row.get('dimensions')[1]
+            # title = row.get('dimensions')[0]
+            # if not title.isascii():
+            #     continue
+            metrics = row.get('metrics')
+
+            for i, values in enumerate(metrics):
+                value = values.get('values')[0]
+                if page in dict:
+                    dict[page] += float(value)
+                    continue
+                dict[page] = float(value)
+
+                pages.append(page)
+
+    # print(dict)
+
+
+average_csats = dict()
+weighted_csats = dict()
+def calculate_csat(no_votes, yes_votes, pageviews):
+
+    # Remove blacklisted pages.
+    for p in blacklist:
+        for k in list(pageviews):
+            if p in k:
+                del pageviews[k]
+        for k in list(no_votes):
+            if p in k:
+                del no_votes[k]
+        for k in list(yes_votes):
+            if p in k:
+                del yes_votes[k]
+
+    # Even out dictionaries.
+    for k,v in pageviews.items():
+        if k not in no_votes:
+            no_votes[k] = 0.0
+        if k not in yes_votes:
+            yes_votes[k] = 0.0
+    for k,v in no_votes.items():
+        if k not in yes_votes:
+            yes_votes[k] = 0.0
+        if k not in pageviews:
+            pageviews[k] = 0.0
+    for k,v in yes_votes.items():
+        if k not in no_votes:
+            no_votes[k] = 0.0
+        if k not in pageviews:
+            pageviews[k] = 0.0
+
+    # Basic CSAT
+    total_yes = 0.0
+    total_no = 0.0
+    for k,v in yes_votes.items():
+        total_yes += v
+    for k,v in no_votes.items():
+        total_no += v
+    total_votes = total_yes + total_no
+    basic_csat = total_yes / total_votes
+
+    # Weighted CSAT
+    for k,v in yes_votes.items():
+        yes = v
+        total = yes + no_votes[k]
+        average_csats[k] = yes / total
```

Comment:
Raises ZeroDivisionError if `yes_votes[k] == 0 and no_votes[k] == 0`. May not be an issue, but it came up in testing.